src/main/scala/org/broadinstitute/dsde/rawls/dataaccess/slick/DriverComponent.scala

```diff
@@ -58,17 +58,22 @@ trait DriverComponent {
     items.zipWithIndex.groupBy(_._2 % batchSize).values.map(_.map(_._1))
   }
 
-  def concatSqlActions(a: SQLActionBuilder, b: SQLActionBuilder): SQLActionBuilder = {
-    SQLActionBuilder(a.queryParts ++ b.queryParts, new SetParameter[Unit] {
+  def concatSqlActions(builders: SQLActionBuilder*): SQLActionBuilder = {
+    SQLActionBuilder(builders.flatMap(_.queryParts), new SetParameter[Unit] {
       def apply(p: Unit, pp: PositionedParameters): Unit = {
-        a.unitPConv.apply(p, pp)
-        b.unitPConv.apply(p, pp)
+        builders.foreach(_.unitPConv.apply(p, pp))
       }
     })
   }
 
-  def concatSqlActionsWithDelim(a: SQLActionBuilder, b: SQLActionBuilder, delim: SQLActionBuilder): SQLActionBuilder = {
-    concatSqlActions(concatSqlActions(a, delim), b)
+  // reduce((a, b) => concatSqlActionsWithDelim(a, b, delim)) without recursion
+  // e.g.
+  //    delim = sql","
+  //    builders = (sql"1", sql"2", sql"3", sql"4")
+  //    output = sql"1,2,3,4"
+  def reduceSqlActionsWithDelim(delim: SQLActionBuilder, builders: Seq[SQLActionBuilder]): SQLActionBuilder = {
+    val elementsWithDelimiters = builders.flatMap(Seq(_, delim)).dropRight(1)
```

Comment:
how about a foldLeft?

builders.tail.foldLeft(Seq(builders.head)) { (elem, accumulator) => accumulator ++ Seq(delim, elem)

not sure that is prettier